lib/models/os.coffee

```diff
@@ -39,24 +239,33 @@ getOsModel = (deps, opts) ->
 	# 	console.log('The raspberry-pi image was last modified in ' + date);
 	# });
 	#
+	# resin.models.os.getLastModified('raspberrypi3', '2.0.0').then(function(date) {
+	# 	console.log('The raspberry-pi image was last modified in ' + date);
+	# });
+	#
 	# resin.models.os.getLastModified('raspberry-pi', function(error, date) {
 	# 	if (error) throw error;
 	# 	console.log('The raspberry-pi image was last modified in ' + date);
 	# });
 	###
-	exports.getLastModified = (deviceType, callback) ->
-		request.send
-			method: 'HEAD'
-			url: "/api/v1/image/#{deviceType}/"
-			baseUrl: imageMakerUrl
-		.catch
-			code: 'ResinRequestError'
-			statusCode: 404
-		, ->
-			throw new errors.ResinRequestError('No such device type')
-		.then (response) ->
-			return new Date(response.headers['last-modified'])
-		.asCallback(callback)
+	exports.getLastModified = (deviceType, version = 'latest', callback) ->
+		callback = findCallback(arguments)
+
+		return isValidDeviceType(deviceType)
+			.then (isValid) ->
+				if not isValid
+					throw new errors.ResinRequestError('No such device type')
+			.then ->
+				return normalizeVersion(version)
+			.then (version) ->
+				imgMakerHelper.request
+					method: 'HEAD'
+					url: "/image/#{deviceType}/?version=#{version}"
+			.catch notFoundResponse, ->
+				throw new errors.ResinRequestError('No such version for the device type')
```

Comment:
Ah, ok, in that case this I'm not that worried. I'm cautious about using the wrong exceptions just because later fixes to those are breaking changes, but if they're already wrong then there's no need to block this on fixing them.